### ISSUE TITLE
Missing documentation newly required autoredirect for auth token config option

### ISSUE THREAD

**Issue by bootc** [docker/distribution#2793]
#2711 adds a new option, `autoredirect`, to the `token` authentication configuration section. This option is _required_ and the registry fails to start without it, but there is no documentation for it whatsoever.

To be honest it feels like it could easily have a default value that maintains the previous behaviour, but either way documentation for it is surely required.

**Comment by caervs** [comment]
@bootc yup. This is a gap in our docs and needs to be addressed. Thank you.

If you're still unsure, `autoredirect` determines whether or not the registry uses the request's Host header when setting the realm for an auth token. Setting this value to `false` will maintain the old behavior.

**Comment by docwhat** [comment]
Why was such a breaking change made in 2.x?

According to [Semantic Versioning](https://semver.org/) it should have required a new major number.

**Comment by donbowman** [comment]
So how does one configure?
There's no REGISTRY_AUTH_TOKEN_AUTOREDIRECT="false" environment variable I can set.

I use the environment variables in my kubernetes chart w/ gitlab to set up AUTH_TOKEN_* and it now won't load.

**Comment by caervs** [comment]
See also #2799 and #2800 which will default the parameter to `false` and docs in #2801. Release plan for these changes TBD.

**Comment by caervs** [comment]
@donbowman 

Yes. `REGISTRY_AUTH_TOKEN_AUTOREDIRECT` should work
